lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -77,7 +69,7 @@ protected function skipClass(ClassMetadata $metadata)
     /**
      * {@inheritDoc}
      */
-    protected function createProxyDefinition($className)
+    protected function createProxyDefinition($className): ProxyDefinition
```

Comment:
@derrabus I can't use a native type declaration here unless we update the parent class first, but as `doctrine/common` still supports 7.1, that would be a BC-break. Should we create branch 4.0.x, drop support for PHP 7 in it, and migrate `doctrine/common` to PHP 8 syntax?

Reply:
no need for major version when dropping 7.1 IMHO

Reply:
Yes, but adding a parameter type hint is a BC-break for callers , and adding a return type declaration is a BC-break for extending classes, so there will be a need for a major release, won't there be?

Reply:
You could add `mixed` as parameter type though.

Reply:
I'll remove ProxyFactory from this PR so that we can move on, let's deal with it later.